### PR TITLE
#62/Rows should conditionally be a link, not just a button

### PR DESCRIPTION
**Pull request by Damrod**

### CODE REVIEW

File: webapp/src/pages/Users/Users.tsx [hunk 1]
```diff
@@ -18,38 +18,35 @@ import {
...
-  username: string;
-  email: string;
-  avatarId: string | null;
-  id: string;
-  createdAt: Date;
+  readonly username: string;
+  readonly email: string;
+  readonly avatarId: string | null;
+  readonly id: string;
+  readonly createdAt: Date;
```
Comment (gbudau) [inline webapp/src/pages/Users/Users.tsx:25]: :+1:

File: webapp/src/pages/Users/Users.tsx [hunk 2]
```diff
@@ -18,38 +18,35 @@ import {
...
-        },
-        title: user.username,
-        subtitle: 'level x',
-        key: user.id,
-      });
-    });
-    return rows;
+    const users = (await fetch(url).then((response) => response.json())) ?? [];
+    return {
+      rows: users.map((user: User) => {
```
Comment (gbudau) [inline webapp/src/pages/Users/Users.tsx:34]: praise: Well done! I like this more.

File: webapp/src/shared/components/Row/Row.tsx [hunk 3]
```diff
@@ -3,10 +3,12 @@ import './Row.css';
 import { TextColor, TextVariant, TextWeight } from '../Text/Text';
 import { default as Text } from '../Text/Text';
 import { AvatarProps, SmallAvatar } from '../Avatar/Avatar';
+import { Link } from 'react-router-dom';
 
 export type RowProps = {
   iconVariant?: IconVariant;
   onClick?: () => void;
+  url?: string;
```
Comment (gbudau) [inline webapp/src/shared/components/Row/Row.tsx:11]: suggestion: Here we could make this component require one of the `url` or `onClick` and disallow both or none.
https://stackoverflow.com/a/61281828

File: webapp/src/shared/components/Row/Row.tsx [hunk 4]
```diff
@@ -15,38 +17,46 @@ export type RowProps = {
...
-            {subtitle}
-          </Text>
-        )}
-      </div>
-      {iconVariant && (
-        <div className="row-icon">
-          {<Icon variant={iconVariant} size={IconSize.SMALL} />}
-        </div>
+  const RowChildren: JSX.Element[] = [
+    (avatarProps && <SmallAvatar {...avatarProps} />) || <div />,
```
Comment (gbudau) [inline webapp/src/shared/components/Row/Row.tsx:26]: suggestion: Here we could use a [fragment](https://reactjs.org/docs/fragments.html#short-syntax). This way, we could omit the empty divs.

**Review by gbudau — APPROVED** [review]
Thank you! LGTM :rocket:

File: webapp/src/shared/components/Row/Row.tsx (hunk 3 above)
Comment (Damrod) [reply webapp/src/shared/components/Row/Row.tsx]: Does this solution work for you?

**Review by fizcamposancos — COMMENTED** [review]
I like it more now

**Pull request merged**